pkg/util/ec2/imds_helpers.go

```diff
@@ -24,15 +24,23 @@ var (
 	imdsNetworkMacs = "/network/interfaces/macs"
 )
 
-// imdsVersion is an enum to determine how to interact with the IMDSv2 option
-type imdsVersion int
+// ec2IMDSVersionConfig is an enum to determine how to interact with the IMDSv2 option
+type ec2IMDSVersionConfig int
 
 const (
-	imdsV1 imdsVersion = iota
+	imdsV1 ec2IMDSVersionConfig = iota
 	imdsAllVersions
 	imdsV2
 )
 
+func (v ec2IMDSVersionConfig) enableV2() bool {
```

Comment:
Maybe `V2Allowed` and `V2Only`? I would assume it's ok to have these functions public as well.

Reply:
yes we can pick these names, I think it makes more sense to me as well than just enable and force

small question, since we are only using these functions inside this file is there a particular interest to make them public ? (it is mainly for my knowledge it is perfectly fine to me to have them as public)